apt/minimization/minimizer.py

```diff
@@ -363,27 +376,90 @@ def fit(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, y: Optional[DATA_PANDA
         :type features_names: list of strings, optional
         :param dataset: Data wrapper containing the training input samples and the predictions of the original model
                         on the training data. Either ``X`` OR ``dataset`` need to be provided, not both.
         :type dataset: `ArrayDataset`, optional
         :return: Array containing the representative values to which each record in ``X`` is mapped, as numpy array or
                  pandas DataFrame (depending on the type of ``X``), shape (n_samples, n_features)
         """
+        transformed = self._inner_transform(X, features_names, dataset)
+        if not self.generalize_using_transform:
+            raise ValueError('transform method called even though generalize_using_transform parameter was False. This '
+                             'can lead to inconsistent results.')
+        transformed_dataset = ArrayDataset(transformed, features_names=self._features)
+        self.calculate_ncp(transformed_dataset)
+        return transformed
+
+    def calculate_ncp(self, samples: Optional[ArrayDataset] = None):
+        """
+        Compute the NCP score of the generalization. Calculation is based on the value of the
+        generalize_using_transform param. If samples are provided, updates stored ncp value to the one computed on the
+        provided data. If samples not provided, returns the last NCP score computed by the `fit` or `transform` method.
+
+        Based on the NCP score presented in: Ghinita, G., Karras, P., Kalnis, P., Mamoulis, N.: Fast data anonymization
+        with low information loss (https://www.vldb.org/conf/2007/papers/research/p758-ghinita.pdf)
+
+        :param samples: The input samples to compute the NCP score on. Ideally should be the data that will be
+                        transformed (e.g., test/runtime data). If not samples supplied, will return the last NCP score
+                        computed by the `fit` or `transform` method.
+        :type samples: ArrayDataset, optional. feature_names should be set.
+        :return: NCP score as float.
+        """
+        if samples is None:
+            return self._ncp
+
+        if not samples.features_names:
+            raise ValueError('features_names should be set in input ArrayDataset.')
+        samples_pd = pd.DataFrame(samples.get_samples(), columns=samples.features_names)
+        if self._features is None:
+            self._features = samples.features_names
+        if self._feature_data is None:
+            self._feature_data = self._get_feature_data(samples_pd)
+        total_samples = samples_pd.shape[0]
+
+        if self.generalize_using_transform:
+            generalizations = self._calculate_cell_generalizations()
+            # count how many records are mapped to each cell
+            counted = np.zeros(samples_pd.shape[0])  # to mark records we already counted
+            ncp = 0
+            for i in range(len(self.cells)):
```

Comment:
Why not iterate over the cells directly? I don't see the index i being used anywhere.